doc/how_to_guides/parallel_planning/src/parallel_planning_main.cpp

```diff
@@ -0,0 +1,300 @@
+#include <moveit/warehouse/trajectory_constraints_storage.h>
+#include <warehouse_ros/database_loader.h>
+
+namespace rvt = rviz_visual_tools;
+
+namespace
+{
+const rclcpp::Logger LOGGER = rclcpp::get_logger("parallel_planning_example");
+const std::string PLANNING_GROUP = "panda_arm";
+}  // namespace
+namespace parallel_planning_example
+{
+/// \brief Find shortest trajectory
+planning_interface::MotionPlanResponse
+getShortestSolution(const std::vector<planning_interface::MotionPlanResponse>& solutions)
+{
+  // Empty line
+  RCLCPP_INFO(LOGGER, "#####################################################");
+  RCLCPP_INFO(LOGGER, "###################### Results ######################");
+  for (auto const& solution : solutions)
+  {
+    RCLCPP_INFO(LOGGER, "Planner '%s' returned '%s'", solution.planner_id_.c_str(),
+                moveit::core::error_code_to_string(solution.error_code_).c_str());
+    if (solution.trajectory_)
+    {
+      RCLCPP_INFO(LOGGER, "Path length: '%f', Planning time: '%f'",
+                  robot_trajectory::path_length(*solution.trajectory_), solution.planning_time_);
+    }
+  }
+  // Find trajectory with minimal path
+  auto const shortest_solution = std::min_element(solutions.begin(), solutions.end(),
+                                                  [](const planning_interface::MotionPlanResponse& solution_a,
+                                                     const planning_interface::MotionPlanResponse& solution_b) {
+                                                    // If both solutions were successful, check which path is shorter
+                                                    if (solution_a && solution_b)
+                                                    {
+                                                      return robot_trajectory::path_length(*solution_a.trajectory_) <
+                                                             robot_trajectory::path_length(*solution_b.trajectory_);
+                                                    }
+                                                    // If only solution a is successful, return a
+                                                    else if (solution_a)
+                                                    {
+                                                      return true;
+                                                    }
+                                                    // Else return solution b, either because it is successful or not
+                                                    return false;
+                                                  });
+  RCLCPP_INFO(LOGGER, "'%s' chosen as best solution (Shortest path)", shortest_solution->planner_id_.c_str());
+  RCLCPP_INFO(LOGGER, "#####################################################");
+  return *shortest_solution;
+}
+
+/// \brief Utility class to create and interact with the parallel planning demo
+class Demo
+{
+public:
+  /// @brief
+  /// @param node
```

Comment:
Either fill these out or remove. For this simple demo you can probably just remove